pkg/sql/sem/builtins/builtins.go

```diff
@@ -4976,17 +4979,29 @@ value if you rely on the HLC for accuracy.`,
 					return nil, errors.AssertionFailedf("expected string value, got %T", args[1])
 				}
 				name := settings.SettingName(strings.ToLower(string(s)))
-				setting, ok, _ := settings.LookupForLocalAccess(name, evalCtx.Codec.ForSystemTenant())
+
+				// To view the per-virtual-cluster overrides for sensitive settings, the
+				// MANAGEVIRTUALCLUSTER privilege is required, since that is the
+				// privilege required to set the override in the first place.
+				hasManage, err := evalCtx.SessionAccessor.HasGlobalPrivilegeOrRoleOption(ctx, privilege.MANAGEVIRTUALCLUSTER)
```

Comment:
It seems a little odd to me to _decode_ a value we're being passed into redaction -- we already had the value when we called the function after all.

I get that this is being done here since this gets called in the internal query, but it seems weird.

What instead of messing with this builtin, and the internal query in sql/tenant_settings.go, we just said that SHOWing any override -- for all overrides regardless of sensitivity -- requires manage tenant rather than viewclustermetadata? would that simplify this?

Reply:
that would simplify things for sure. it makes sense to me too, since (unintentionally or not), `SHOW CLUSTER SETTINGS FOR VIRTUAL CLUSTER ...` already required `MANAGEVIRTUALCLUSTER`, but `SHOW CLUSTER SETTING <name> FOR VIRTUAL CLUSTER` did not.

i will make this change